content/developer/howtos/frontend_owl_components.rst

```diff
@@ -0,0 +1,187 @@
+            </xpath>
+        </template>
+    </odoo>
+
+Don't forget to add this file to the data section of your assets bundle:
+
+.. code-block:: py
+
+    {
+        # ...
+        'data': {
+            'views/templates.xml',
+        }
+    }
+
+And that's it! If you open the home page of the portal you should see the message
+"Hello, World!" at the top of the page.
+
+Problems with using Owl components in the portal and website
+============================================================
+
+Owl components are rendered entirely in JavaScript by the browser. This can cause
+some issues:
+
+- Layout shift
+- Poorer indexing by search engines
+
+For these reasons, you should only use Owl components in the portal and website for
+specific use cases described below.
+
+Layout shift
+------------
+
+When a page initially renders content, and that content subsequently moves ("shifts")
+within the page, this is refered to as a layout shift. When using Owl components in
+the portal or website, all of the HTML that surrounds the Owl component is rendered
+by the server, and is the first thing that will be displayed to the user. When JavaScript
+starts running, Owl will mount your component, which will likely cause the surrounding
+elements to move around in the page. This can cause poor user experience: the user sees
+and element in the page that was initially rendered and with which they want to interact,
+so they move their cursor or finger above that element. Just as they're about to click,
+the Owl component is mounted and the element with which they wanted to interact moved.
+They click and interact with the Owl app instead.
+
+This can be a frustrating experience, so you should be careful when designing your
+page that the Owl component will not move elements around. This can be achieved in
+various ways, e.g. by positioning it below all other existing elements, by not having
+other interactive elements around, or by reserving a fixed space for the Owl component
+using CSS.
+
+
+.. seealso::
+    `cumulative layout shift on web.dev <https://web.dev/articles/cls>`_
+
+
+Poorer indexing by search engines
+---------------------------------
+
+When search engines build their index of the content of the web, they use web crawlers
+to find pages and analyze their content so that they can show these pages in their
+search results. While modern search engines are generally capable of executing JavaScript
+code, and should generally be able to view and index content rendered in JavaScript,
+they may not index the content as fast and may penalize the page in search results.
+
+Because most search engines do not reveal the exact way that they crawl and index
+web pages, it's not always easy to know the extent of the impact client side rendering
+can have on your search engine scores. While it's unlikely to make or break your SEO
+strategy, you should still only use Owl components when they are adding real value
+over server-side rendering.
```

Comment:
```suggestion
When search engines build their index of the content of the web, they use web crawlers
to find pages and analyze their content to show these pages in their
search results. While modern search engines are generally capable of executing JavaScript
code and should generally be able to view and index content rendered in JavaScript,
they may not index the content as fast and penalize the page in search results.

Because most search engines do not reveal the exact way that they crawl and index
web pages, it's not always easy to know the extent of the impact that client-side rendering
can have on your search engine scores. While it's unlikely to make or break your SEO
strategy, you should still only use Owl components when they are adding real value
over server-side rendering.
```